### ISSUE TITLE
Features list ?

### ISSUE THREAD

**Issue by vzapo** [RetroAchievements/RABot#42]
Hey guys,

I looked over the bot a bit and other issues/suggestions and thought about the following things to implement next:

1. Bot announcing different winners of different events. 
 - I would think to implement it one of the 2 following ways:
 1.1. If done automatically presumably it would have to pull some endpoint data on the event on a time basis
 1.2.  to do it daily once, at given time an announcement with the players who won in the current day
Both ways would need a feed from which I could import it,if any of you guys could point out where i could get the info, would be grateful :)

2. Not sure it is worth but a kick/ban command from discord and also from site ? if it's worth would have thought to implement it something like:
 2.1. for ban for example `!ban discorduser websiteuser`
 2.2. for kick `!kick discorduser`

3. Might worth implementing a mute command for people that do not behave and have some sort of warning before getting kick/ban ? 
 3.1. to implement a mute thing, i was thinking to create a mute role that has no text permissions which the bot will add
 3.2. to unmute it can add a timer or a manual unmute  must be done, one of the 2... or some other idea

From other issues/features open I could also add the following:
- implementing different/multi prefix. This is not really feasible in the current state using Commando Framework. i tried a hacky way to implement this by catching the command before it was executed but it ended up to ugly to implement in the long run and didn't worked as expected...
this is due to the fact that Commando framework implements the prefix automatically with `! ` if not defined and if it is defined will accept only plain strings,not even regex can work... might worth look into it later(#1)
- giving the bot some AI :) (#29). This could be done in multiple ways, but one i guess one easy way would be to add Dialogflow form Google or maybe other one like IBM Watson or Amazon LEX?... would incline thou for Dialogflow it;'s a bit more easy to use in my opinion...
- hidden vote concept(#30) I could also work on this if it's still needed and not done 
-  verify @mention to verify new users (only for mods) & setaotw @mention to give "aotw winner" 
 (#1) I could also do this but would need somebody to tell me where can I make a request for this (the endpoint that is accepting it?)

If you think any of this sounds good or have anything else, any other ideas I'm up for it... If I can handle it :)

**Comment by meleu** [comment]
**TL;DR**: I've just edited #1 in a hope to make it RABot's roadmap.

---

> 1. Bot announcing different winners of different events

As a priority I would like to see the aotw winners being announced in the #aotw channel (no need to `@mention` anyone, just say their names).

My suggestion is to check `API_GetAchievementOfTheWeek.php` each 30min (96 requests/day, looks pretty soft to me).

Events that are not ~~official~~ coded in the backend would be trickier to do, and IMHO it doesn't worth the effort.


> 2. (...) kick/ban command (...)
> 3. (...) mute command (...)

I was interested on it in the past, but right now I don't see it as a priority.

Currently admins/mods are using [Carl-bot](https://carl.gg/) to help with the moderation tasks and it seems to be doing a good job so far.

Yeah, I don't like the fact it's not opensource, but we have a lot of other concerns on the "things to worry" queue. :sweat_smile:


> implementing different/multi prefix.

I don't see a reason for that. Actually I'm against this idea. Using `!` is enough.


> giving the some AI

Wow! It always appeared to me that things in #29 don't need to be that fancy. Just some logic with time and such.


> hidden vote concept 

Oh sure. It seems to be easy if making it act like a [tpoll](https://github.com/RetroAchievements/RABot/blob/master/commands/poll/tpoll.js) and removing user's vote right after they add a reaction.


> verify @mention to verify new users (only for mods) & setaotw @mention to give "aotw winner"

the verify command would be useful, indeed.

the setaotw/purgeaotw one would surely help mods and aotw maintainers to manage those roles.
